### ISSUE TITLE
Encountered Error "C4146: unary minus operator applied to unsigned type, result still unsigned"

### ISSUE THREAD

**Issue by rapoth** [lemire/EWAHBoolArray#16]
Hi - I am attempting to compile some examples using this library in Visual Studio 2015 (compiling for x64). While all your example work perfectly, I am having trouble using any of the to* methods (i.e., toArray(), toVector() etc.) as the compile throws an error:

> "unary minus operator applied to unsigned type, result still unsigned"

For now, I am able to retrieve the row offsets using the iterator (code below) you provided in your example but I also saw your comment inside the source that this might be inefficient and that I should use toArray when possible.

```
for (EWAHBoolArray<uint32_t>::const_iterator j = bitmap.begin(); j != bitmap.end(); ++j) {
        std::cout << "term '" << key << "' appears at row index " << *j << std::endl;
}
```

Do you have any suggestions on how I can resolve this problem?

The exact line of code that is leading to this error is the following:

```
template <class uword>
std::vector<size_t> EWAHBoolArray<uword>::toArray() const {
      ...
      ...
      while (myword != 0) {
        uint64_t t = myword & -myword; // <--------------------- Compiler complaining here
        uint32_t r = numberOfTrailingZeros(t);
        ans.push_back(pos + r);
        myword ^= t;
      }
      ... 
      ...
  return ans;
}
```

Thank you!

**Comment by lemire** [comment]
@rapoth 

Thanks.

Is it a compiler warning (as in: the code still build) or a compiler error (you need to comment out the whole function for the code to build)?

**Comment by rapoth** [comment]
Sorry for the delay. I should have clarified. The code from the repo compiles just fine in Visual Studio 2015.  The problem occurs when I use the toVector() method in my code and then include the ewah.h header. And from what I am noticing this is not a warning but a compiler error (most likely because of some configuration or a bug in VS):

![image](https://cloud.githubusercontent.com/assets/8262876/18731559/74cc80d8-8011-11e6-93fa-9232fe4c0975.png)

I found two ways this can be fixed:

1. Use #pragma disable in ewah.h 

> #pragma warning (disable : 4146)

2. Pass additional flags in VS for my project **/wd4146**

![image](https://cloud.githubusercontent.com/assets/8262876/18733477/26df2d76-8020-11e6-9404-91c19a9479fe.png)

Thanks again!

**Comment by lemire** [comment]
@rapoth  The screenshot you are showing seems to indicate that there is a warning. Not an error.

**Comment by rapoth** [comment]
@lemire Nope. In my VS, the red cross indicates an error. Warnings are shown in yellow (example below):

![image](https://cloud.githubusercontent.com/assets/8262876/18733639/959dbee8-8021-11e6-9aa6-cac65079fb48.png)

**Comment by lemire** [comment]
My understanding of Visual Studio is that errors are reported in the Output window. Can you provide me with a copy of the Output window where the "errors" appear? I bet that you see no such error in the Output window (though you might see warnings).

The screenshot you are providing looks to me like an Intellisense report, not a compiler report. 

I submit to you that the "error" is not a compiler error at all. We can check it out easily by disabling Intellisense error reporting. Go to``Tools -> Options -> Text Editor -> C/C++ -> Advanced``, and set ``Disable Error Reporting`` to True. 

You can also install a plugin...

https://visualstudiogallery.msdn.microsoft.com/7c43abba-08ad-4b92-aa83-710ff9f9bd90?SRC=Home

**Comment by rapoth** [comment]
@lemire I changed the option to show only Build errors to ensure that we are not seeing any intellisense errors. Please look at the new screenshot below. I also tried setting ```Disable Error Reporting``` to true but that did not make any difference.

![image](https://cloud.githubusercontent.com/assets/8262876/18734064/72a0c9d6-8025-11e6-9c5a-434c4316809e.png)

Here's the relevant text from the Output window:

```
> ...
> 3>E:\github\Example1\thirdparty\EWAHBoolArray/headers/ewah.h(965): error C4146: unary minus operator applied to unsigned type, result still unsigned
> 3>  E:\github\Example1\thirdparty\EWAHBoolArray/headers/ewah.h(944): note: while compiling class template member function 'std::vector<size_t,std::allocator<_Ty>> EWAHBoolArray<uint32_t>::toArray(void) const'
> 3>          with
> 3>          [
> 3>              _Ty=size_t
> 3>          ]
> ..
> ..
> ========== Rebuild All: 2 succeeded, 1 failed, 0 skipped ==========
```

Please let me know if there's any other information you need from my end. Thanks again!

**Comment by lemire** [comment]
Does the error remains if you upgrade to the latest version...

https://github.com/lemire/EWAHBoolArray/releases/tag/v0.5.8

**Comment by rapoth** [comment]
@lemire No, it went away. I am able to successfully compile after updating to the latest version. Many thanks for your time and patience!

**Comment by lemire** [comment]
Thanks for your help.
